input/cql/HIVCommon.cql

```diff
@@ -743,161 +184,118 @@ define function ToDaily(frequency System.Integer, period System.Quantity):
-        when difference in days between start of "First HIV Treatment".effective.toInterval() and start of "First HIV Positive Condition".onset.toInterval() <= 30 then HC."Within 30 days of HIV diagnosis"
-        when difference in days between start of "First HIV Treatment".effective.toInterval() and start of "First HIV Positive Condition".onset.toInterval() <= 90 then HC."Within 90 days of HIV diagnosis"
-    else null
-    end
+    and exists(O.category OC where OC ~ CustomConcepts."social-history")
+    and O.code ~ Concepts."Sexual partner - HIV.B.DE11" 
   
   //HIV status of partner or contact (already known positive, newly diagnosed positive, negative) 
 
   define "already known positive":
-    [Observation: HC."HIV status of partner or contact"] O 
+    [Observation: Concepts."HIV status of partner or contact"] O 
     where O.status in { 'final', 'amended', 'corrected' }
-    and exists(O.category OC where OC ~ HCC."social-history")
-    and O.code ~ HC."Already knew positive"
+    and exists(O.category OC where OC ~ CustomConcepts."social-history")
+    and O.code ~ Concepts."Already knew positive"
   
   define "Newly diagnosed":
-    [Observation: HC."HIV status of partner or contact"] O 
+    [Observation: Concepts."HIV status of partner or contact"] O 
     where O.status in { 'final', 'amended', 'corrected' }
-    and exists(O.category OC where OC ~ HCC."social-history")
-    and O.code ~ HC."Newly diagnosed"
+    and exists(O.category OC where OC ~ CustomConcepts."social-history")
+    and O.code ~ Concepts."Newly diagnosed"
 
   define "Negative - HIV.H.DE37":
-    [Observation: HC."HIV status of partner or contact"] O 
+    [Observation: Concepts."HIV status of partner or contact"] O 
     where O.status in { 'final', 'amended', 'corrected' }
-    and exists(O.category OC where OC ~ HCC."social-history")
-    and O.code ~ HC."Negative - HIV.H.DE37"
+    and exists(O.category OC where OC ~ CustomConcepts."social-history")
+    and O.code ~ Concepts."Negative - HIV.H.DE37"
   
 
 
 //need to define
   define STI_testing:
     [Procedure] P
   where P.status = 'completed'
-  and P.code ~ HC."STI testing and treatment services"
+  and P.code ~ Concepts."STI testing and treatment services"
+
+
+/**
+ * @description Takes a choice between a Medication and a CodeableConcept and returns just the code of the medication
+ */
+define function ExtractMedicationCode(choice Choice<FHIR.CodeableConcept, FHIR.Reference>):
+  case
+	  when choice is FHIR.CodeableConcept then
+    	choice as FHIR.CodeableConcept
+    when choice is FHIR.Reference then
+      First([Medication] M 
+        where M.id = Last(Split(choice.reference, '/'))
+        return M.code as FHIR.CodeableConcept)
+		else
+      Message(null as FHIR.CodeableConcept, true, '1', 'Error', 'Cannot compute a medication code') // TODO: I'm sure that this is supported somehow?
+	end
```

Comment:
nit: this element uses both spaces and tabs, which means the indentations renders inconsistently.